src/components/NewCommentForm.tsx

```diff
@@ -1,6 +1,160 @@
-import React from 'react';
+import React, { FormEvent, useState } from 'react';
+import { addNewComment } from '../services/comment';
+import classNames from 'classnames';
+import { Comment, CommentData } from '../types/Comment';
+
+type Props = {
+  postId: number;
+  setComments: React.Dispatch<React.SetStateAction<Comment[]>>;
+  setError: (value: boolean) => void;
+};
+
+export const NewCommentForm: React.FC<Props> = ({
+  postId,
+  setComments,
+  setError,
+}) => {
+  const [newComment, setNewComment] = useState<CommentData>({
+    name: '',
+    email: '',
+    body: '',
+  });
+
+  const [newCommentErrors, setNewCommentErrors] = useState({
+    name: false,
+    email: false,
+    body: false,
+  });
+  const [isLoading, setIsLoading] = useState(false);
+
+  function handleOnChange(
+    event: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>,
+  ) {
+    if (event.target.name === 'name') {
+      setNewComment(prev => {
+        return {
+          ...prev,
+          [event.target.name]: event.target.value,
+        };
+      });
+
+      setNewCommentErrors(prev => {
+        return {
+          ...prev,
+          [event.target.name]: false,
+        };
+      });
+    }
+
+    if (event.target.name === 'email') {
+      setNewComment(prev => {
+        return {
+          ...prev,
+          email: event.target.value,
+        };
+      });
+
+      setNewCommentErrors(prev => {
+        return {
+          ...prev,
+          email: false,
+        };
+      });
+    }
+
+    if (event.target.name === 'body') {
+      setNewComment(prev => {
+        return {
+          ...prev,
+          body: event.target.value,
+        };
+      });
+      setNewCommentErrors(prev => {
+        return {
+          ...prev,
+          body: false,
+        };
+      });
+    }
```

Comment:
```suggestion

      setNewComment(prev => {
        return {
          ...prev,
          [event.target.name]: event.target.value,
        };
      });

      setNewCommentErrors(prev => {
        return {
          ...prev,
          [event.target.name]: false,
        };
      });
```
you can simplify that